gnovm/pkg/gnolang/alloc.go

```diff
@@ -12,6 +12,10 @@ type Allocator struct {
 	maxBytes int64
 	bytes    int64
 	collect  func() (left int64, ok bool) // gc callback
+
+	// For memory profiling
+	machine   *Machine
+	allocType string // Current allocation type being tracked
```

Comment:
I tried to modify the Alloc functions to pass a type string to `allocType` field and then replace it with an empty string, but this caused nil pointer errors when empty strings were used, so I've left it as a placeholder for now.